src/Scheduler.scala

```diff
@@ -0,0 +1,20 @@
+import java.util.concurrent.{Future, PriorityBlockingQueue}
+
+import scala.concurrent.ExecutionContext
+
+trait Scheduler {
+  /**
+    * should include the queue as a property like class?
+    */
+  //  val minHeap: PriorityBlockingQueue[(Task, Long)]
+  def schedule(task: Executable, timestamp: Long): Unit
+  def start()(implicit ec: ExecutionContext): Unit
+}
+
+/**
+  * Choose among trait, object, class, or case class?
```

Comment:
I think class is most appropriate, since there's no reason to use pattern matching or serialization on it, and in most cases there should only be one instance. Anything *could* be a trait in Scala, which would allow the methods and functionality to be mixed in with another class.. but you'd have to ask yourself why? It probably makes more sense to declare scheduler as a member of another class (or singleton object), instead of mixing-in the functionality as a trait.